tests/integration/test_api.py

```diff
@@ -386,3 +386,47 @@ def test_annotation_event(client, monkeypatch, httpserver):
             },
         )
     assert result.status_code == 200
+
+def test_annotate_insight_with_user_id(client):
+    # to test annotation insight when the user is authenticated and less votes are provided
+    result = client.simulate_post(
+        "/api/v1/insights/annotate",
+        params={
+            "insight_id": insight_id,
+            "annotation": -1,
+            "device_id": "voter1",
+        },
+    )
+
+    assert result.status_code == 200
+    assert result.json == {"description": "the annotation was saved", "status": "saved"}
+
+    
+    authenticated_result = client.simulate_post(
+        "/api/v1/insights/annotate",
+        params={
+            "insight_id": insight_id,
+            "annotation": -1,
+            "device_id": "voter1",
+        },
+        headers={"Authorization": "Basic " + base64.b64encode(b"a:b").decode("ascii")},
+    )
+
+    assert authenticated_result.status_code == 200
+    assert authenticated_result.json == {"description": "the annotation was saved", "status": "saved"}
+    # For authenticated users we expect the insight to be validated directly, tracking the username of the annotator.
+    votes = list(AnnotationVote.select())
+    assert len(votes) == 0
+
+    insight = next(
+        ProductInsight.select()
+        .where(ProductInsight.id == insight_id)
+        .dicts()
+        .iterator()
+    )
+    assert insight.items() > {"username": "a", "annotation": -1, "n_votes": 0}.items()
```

Comment:
I believe I should increase the count of `"n_votes"` to 1 because the user has already voted once